controllers/notification/notification_builder_test.go

```diff
@@ -25,6 +25,24 @@ func TestNotificationBuilder(t *testing.T) {
 		require.Equal(t, "foo@acme.com", notification.Spec.Recipient)
 	})
 
+	t.Run("test create fails with empty email address", func(t *testing.T) {
+		// when
+		_, err := NewNotificationBuilder(client, test.HostOperatorNs).Create("")
+
+		// then
+		require.Error(t, err)
+		require.Equal(t, "The specified recipient [] is not a valid email address", err.Error())
+	})
+
+	t.Run("test create fails with invalid email address", func(t *testing.T) {
+		// when
+		_, err := NewNotificationBuilder(client, test.HostOperatorNs).Create("foo")
+
+		// then
+		require.Error(t, err)
+		require.Equal(t, "The specified recipient [foo] is not a valid email address", err.Error())
+	})
```

Comment:
The email regex looks quite complicated. Can we add a bunch of tests which would test different email formats like:
- `John Wick <johnWick@domain.com>`, `john.wick@subdomain.domain.com`, `john-Wick@domain.com`, etc
vs different invalid emails

Reply:
We don't store any email addresses in your first format.  I've added tests for the other two though.